api/provider/workos.go

```diff
@@ -0,0 +1,84 @@
+package provider
+
+import (
+	"context"
+	"strings"
+
+	"github.com/netlify/gotrue/conf"
+	"golang.org/x/oauth2"
+)
+
+const (
+	defaultWorkOSAPIBase = "https://api.workos.com"
+)
+
+type workosProvider struct {
+	*oauth2.Config
+	APIPath string
+}
+
+type workosUser struct {
+	ID             string                 `json:"id"`
+	ConnectionId   string                 `json:"connection_id"`
+	ConnectionType string                 `json:"connection_type"`
+	Email          string                 `json:"email"`
+	FirstName      string                 `json:"first_name"`
+	LastName       string                 `json:"last_name"`
+	Object         string                 `json:"object"`
+	IdpId          string                 `json:"idp_id"`
+	RawAttributes  map[string]interface{} `json:"raw_attributes"`
+}
+
+// NewWorkOSProvider creates a WorkOS account provider.
+func NewWorkOSProvider(ext conf.OAuthProviderConfiguration, scopes string) (OAuthProvider, error) {
+	if err := ext.Validate(); err != nil {
+		return nil, err
+	}
+
+	apiPath := chooseHost(ext.URL, defaultWorkOSAPIBase)
+
+	oauthScopes := []string{}
+
+	if scopes != "" {
+		oauthScopes = append(oauthScopes, strings.Split(scopes, ",")...)
+	}
+
+	return &workosProvider{
+		Config: &oauth2.Config{
+			ClientID:     ext.ClientID,
+			ClientSecret: ext.Secret,
+			Endpoint: oauth2.Endpoint{
+				AuthURL:  apiPath + "/sso/authorize",
```

Comment:
Some issues with the workos api, i tried setting up an account and faced with this message after calling `/sso/authorize`.

I tried adding the domain query param manually and it seemed to work so maybe its not optional even though it says so in their api. I've already gave my feedback to workos, hopefully they offer a workaround of some sort.

![image](https://user-images.githubusercontent.com/28647601/143217566-cc9d6f5b-0c2f-48b3-b179-1a4965d8d7c7.png)

Reply:
Hey @kangmingtay, did the WorkOS team ever get back to you about your feedback?

Reply:
hmm i haven't heard back from them yet, will reach out again to check!